### ISSUE TITLE
Possible to specify a newer version of prettier?

### ISSUE THREAD

**Issue by billkoch** [diffplug/spotless#362]
First off, thanks for all the hard work that has gone into this project!

I was looking to use [the support for Prettier](https://github.com/diffplug/spotless/tree/master/plugin-gradle#applying-prettier-to-javascript--flow--typescript--css--scss--less--jsx--graphql--yaml--etc) to format YAML files in my project.

With the following configuration:
```groovy
spotless {
  format 'yaml', {
    target '**/*.yml', '**/*.yaml'
    prettier().config(['parser': 'yaml'])
  }
}
```

I was getting:
```
Error: Couldn't resolve parser "yaml"
    at _construct (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:1922:22)
    at _construct (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:1928:21)
    at ConfigError.Wrapper (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:1948:14)
    at new ConfigError (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:2027:74)
    at resolveParser$1 (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:6972:13)
    at normalize (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:7063:19)
    at formatWithCursor (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:10372:12)
    at /Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:31126:15
    at Object.format (/Users/kochwm/dev/istio-demo/java/build/spotless-node-modules-prettier-format/node_modules/prettier/index.js:31145:12)
```

After doing some digging, I noticed [Prettier doesn't support YAML until version 1.14.0](https://prettier.io/docs/en/options.html#parser) and (if I understand how things work correctly) [prettier-package.json is pulling in 1.13.4](https://github.com/diffplug/spotless/blob/master/lib/src/main/resources/com/diffplug/spotless/npm/prettier-package.json#L5)

Is it possible to specify a newer version of prettier in my spotless configuration?

It's also possible (read: likely) I'm barking up the wrong tree entirely, so any tips/suggestions would be greatly appreciated.

Thanks!

**Comment by nedtwigg** [comment]
Thanks for the bug @billkoch!  I think you're barking up exactly the right tree.  Our documentation says that we support YAML via prettier, but we are hardcoded to version 1.13.4, which does not support YAML.

It looks to me like it would be pretty easy to fix this as such:

- make `prettier().config(...)` into `prettier('version').config`
- update [this](https://github.com/diffplug/spotless/blob/009a2ffc6b3f0a469288be84f98cddb1388cd57d/lib/src/main/java/com/diffplug/spotless/npm/PrettierFormatterStep.java#L55-L56) to support that change

@simschla is there any hidden knowledge here that's important?  Are we stuck on `1.13.4` because it's the only one that worked with J2V8, for example?

**Comment by simschla** [comment]
@nedtwigg I think the suggested approach to use a `prettier('version')`would fit perfectly with other cases such as the eclipse or google formatter 👍 
(It might also be a good idea to bump the version numbers as far up as possible when introducing this change on master)

Regarding version-possibilities. We are currently limited by J2V8:
The Version of J2V8 limits the available node and npm version ... and this might or might not limit the supported prettier-version(s). 

Prettier requires a node version `>= 6` according to their [package.json](https://github.com/prettier/prettier/blob/db560e4e1916366d2da739f90a5a2aa0f198432c/package.json#L14). 

I think node version 6 is supported by the currently used version of J2V8, so bumping the prettier dependency should be no problem.
Maybe it will come a time when we will need to [build our own](https://github.com/eclipsesource/J2V8/blob/master/BUILDING.md) J2V8 versions?

**Comment by nedtwigg** [comment]
> Maybe it will come a time when we will need to [build our own](https://github.com/eclipsesource/J2V8/blob/master/BUILDING.md) J2V8 versions?

I'd prefer to help J2V8 update their published artifacts rather than publish our own.  In the meantime, it's good to know that as of prettier `1.17.0-dev`, there's no problem with our current J2V8 that's stuck on node 6.

So the path forward to fixing this in at least the medium term is clear and pretty easy, the only question is when someone will find the time to do it :)  It's not gonna make the top of my todo list anytime soon, but I'll be sure to quickly merge and release anyone's efforts.

**Comment by simschla** [comment]
> but I'll be sure to quickly merge and release anyone's efforts

I'll give it a go, and see how far I get!

**Comment by nedtwigg** [comment]
Released in 3.20.0 / 1.20.0
